src/lsp/cobol_parser/expect/gen_expect.ml

```diff
@@ -167,234 +175,238 @@ module CompEntrySet = Set.Make(struct
   let rec continue_while_terminal rhs i l =
     match rhs.(i) with
-        | T t, _, _ when is_valid_for_comp t ->
-            continue_while_terminal rhs (i+1) (t::l)
-        | _ | exception Invalid_argument _ -> List.rev l
+    | T t, _, _
+      when is_valid_for_comp t ->
+      continue_while_terminal rhs (i+1) (t::l)
+    | _
+    | exception Invalid_argument _ ->
+      List.rev l
   in
   let next_tokens_from_item item =
     let (prod, idx) = item in
     let rhs = Production.rhs prod in
     match rhs.(idx) with
-      | N nt, _, _ -> List.filter_map terminal_filter_map (Nonterminal.first nt)
-      | T _, _, _ -> begin
+    | N nt, _, _ -> List.filter_map terminal_filter_map (Nonterminal.first nt)
+    | T _, _, _ -> begin
         match continue_while_terminal rhs idx [] with
-          | [] -> []
-          | l -> [K (NEL.of_list l)] end
-      | exception Invalid_argument _ -> []
+        | [] -> []
+        | l -> [K (NEL.of_list l)] end
+    | exception Invalid_argument _ -> []
   in
   Lr1.tabulate begin fun lr1 ->
     let custom_comp_entries = CompEntrySet.from_list_map
-      ~add:CompEntrySet.add_option
-      begin function
-        | N nonterm, _ -> nonterminal_filter_map nonterm
-        | _ -> None end
-      (Lr1.transitions lr1) in
+        ~add:CompEntrySet.add_option
+        begin function
+          | N nonterm, _ -> nonterminal_filter_map nonterm
+          | _ -> None end
+        (Lr1.transitions lr1) in
     CompEntrySet.from_list_map
       ~init:custom_comp_entries ~add:CompEntrySet.add_list
       next_tokens_from_item (Lr0.items @@ Lr1.lr0 lr1)
   end
 
 let emit_acceptable_terminals_in_env ppf = (* taking transitions *)
-  Fmt.pf ppf {|
+  Fmt.string ppf {|
 let acceptable_terminals_in ~env : completion_entry list =
   NEL.(match Menhir.current_state_number env with
 |};
   Lr1.fold (fun lr1 acc ->
-    match completion_entries_of lr1 with
+      match completion_entries_of lr1 with
       | s when CompEntrySet.is_empty s -> acc
       | s -> CompEntryMap.add_to_list s lr1 acc)
-  CompEntryMap.empty
+    CompEntryMap.empty
   |> inv (CompEntryMap.fold (fun s l acc -> (l, CompEntrySet.elements s)::acc)) []
   |> pp_match_cases ppf
     Fmt.(using Lr1.to_int int)
     (pp_brackets @@ pp_list pp_completion_entry)
     "[])"
 
 let guess_default_value ~mangled_name ~typ =
-  match typ, mangled_name with
-  | _ when EzString.ends_with ~suffix:"option" typ ||
-          EzString.starts_with ~prefix:"option_" mangled_name
-  -> Some "None"
-  | _ when EzString.ends_with ~suffix:"list" typ ||
-  EzString.starts_with ~prefix:"loption_" mangled_name ||
-  EzString.starts_with ~prefix:"list_" mangled_name
-  -> Some "[]"
-  | _ when EzString.ends_with ~suffix:"unit" typ
-  -> Some "()"
-  | _ when EzString.ends_with ~suffix:"bool" typ ||
-  EzString.starts_with ~prefix:"boption_" mangled_name
-  -> Some "false"
-  | _ -> None
+  if EzString.ends_with ~suffix:" option" typ ||
```

Comment:
So here you don't deal with `int * int option`. Where is this handled?

Reply:
I added a `@default`, it was the only case in the grammar with a product type ending with option:
```ocaml
from_to_characters_opt [@default (None, None)]:
```

Reply:
I'm curious as to what happens if you remove that `@default` annotation. Does the `gen_expect.ml` script fail? or the compilation of `grammar_expect.ml`? a potential error at runtime (upon a completion request)? or just some completion cases that don't show up?

Reply:
In this case the `grammar_expect.ml` fails to compile because some `match` case is misstyped:
```shell
File "src/lsp/cobol_parser/grammar_expect.ml", line 3430, characters 32-36:
3430 |   | N_from_to_characters_opt -> None
                                       ^^^^
Error: This expression should not be a constructor, the expected type is a
```

Reply:
Ok, good to known it's a static check.